internal/controller/model_controller_test.go

```diff
@@ -9,13 +9,24 @@ import (
 	apiv1 "github.com/substratusai/substratus/api/v1"
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/util/retry"
 	"k8s.io/utils/ptr"
 )
 
+func RetryOnConflictError(operation func() error) error {
+	backoff := retry.DefaultBackoff
+	backoff.Steps = 5
+
+	return retry.OnError(backoff, func(err error) bool {
+		return errors.IsConflict(err)
+	}, operation)
+}
```

Comment:
We don't have a pattern established just yet of where to put common test utilities like this. I could see a `testutil` package becoming useful but we can probably hold off until we see it (or similar functions) reused.